sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -41,7 +42,8 @@ void Scheduler::GraphProcessor::waitForEvent(EventImplPtr Event) {
   assert(Cmd && "Event has no associated command?");
   EnqueueResultT Res;
   bool Enqueued = enqueueCommand(Cmd, Res, BLOCKING);
-  if (!Enqueued && EnqueueResultT::SyclEnqueueFailed == Res.MResult)
+  if (!Enqueued && EnqueueResultT::SyclEnqueueFailed == Res.MResult &&
+      Res.MErrCode != CL_SUCCESS)
```

Comment:
I don't like this CL_SUCCESS here but I need it.
Somehow I need to understand the command failure reason: it was because of one of its dependencies failed while executing, or it was because of one of its dependencies failed earlier.
Does it make sense to add a new field into EnqueueResultT structure?